src/corehost/cli/ijwhost/corhdr.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+#ifndef CORHDR_H
+#define CORHDR_H
+
+#include "pal.h"
+
+typedef struct IMAGE_COR_VTABLEFIXUP // From CoreCLR's corhdr.h
+{
+    std::uint32_t       RVA;                    // Offset of v-table array in image.
+    USHORT      Count;                  // How many entries at location.
+    USHORT      Type;                   // COR_VTABLE_xxx type of entries.
+} IMAGE_COR_VTABLEFIXUP;
+
+#define RidFromToken(tk) ((ULONG32) ((tk) & 0x00ffffff))
+#define TypeFromToken(tk) ((ULONG32)((tk) & 0xff000000))
+#define IsNilToken(tk) ((RidFromToken(tk)) == 0)
+
+
+using mdToken = std::uint32_t;
+
+typedef enum CorTokenType
```

Comment:
Are these token definitions really only used for asserts?

Reply:
On x86 it looks like the stub encodes it in whether or not it is a mdtMethodDef or mdtMemberRef. I don't see this encoding being used anywhere though so I can remove it, and also remove the asserts if you would like.

Reply:
Yes, I think it would be nice to make this code as simple as possible, depending on as few things as possible.